lib/Optimizer/Transforms/DependencyAnalysis.cpp

```diff
@@ -653,6 +855,9 @@ class OpDependencyNode : public DependencyNode {
     dependencies = _dependencies;
 
     quantumOp = isQuakeOperation(op);
+    // TODO: quake.discriminate is currently the only operation in the quake
+    // dialect that doesn't operate on wires. This will need to be updated if
+    // that changes.
```

Comment:
This isn't strictly true. For example, `quake.alloca` doesn't operate on wires either, but I suspect you aren't expecting to see those in this pass?